content/src/content/jcr_root/apps/core/wcm/components/image/v2/image/clientlibs/site/js/image.js

```diff
@@ -283,11 +333,19 @@
         function cacheElements(wrapper) {
             that._elements = {};
             that._elements.self = wrapper;
-            var hooks = that._elements.self.querySelectorAll("[data-" + NS + "-hook-" + IS + "]");
-
-            for (var i = 0; i < hooks.length; i++) {
-                var hook = hooks[i];
-                var capitalized = IS;
+            if (!that._properties.isBgImage) {
+                var hooks = that._elements.self.querySelectorAll("[data-" + NS + "-hook-" + IS + "]");
+
+                for (var i = 0; i < hooks.length; i++) {
+                    var loopHook = hooks[i];
+                    var loopCapitalized = IS;
+                    loopCapitalized = loopCapitalized.charAt(0).toUpperCase() + loopCapitalized.slice(1);
+                    var loopKey = loopHook.dataset[NS + "Hook" + loopCapitalized];
+                    that._elements[loopKey] = loopHook;
+                }
+            } else {
+                var hook = that._elements.self;
```

Comment:
if we just do `var hooks = [that._elements.self];` and `var capitalized = BG;` for the case of a BG image, then the rest of the code in this function can stay the exact same as it was before instead of copy/pasting it.

Reply:
also, any reason why the image version needs to query for multiple elements, but the BG version of this code does not?